pkg/kubepug/input/k8s/deleted.go

```diff
@@ -138,36 +140,24 @@ func GetDeleted(kubeAPIs parser.KubernetesAPIs, config *genericclioptions.Config
 
 	log.Debugf("Walking through %d resource types", len(resourcesList))
 	for _, resourceGroupVersion := range resourcesList {
-		// We dont want CRDs or APIExtensions to be walked
+		// We don't want CRDs or APIExtensions to be walked
 		if _, ok := ignoreObjects[strings.Split(resourceGroupVersion.GroupVersion, "/")[0]]; ok {
 			continue
 		}
 
 		for i := range resourceGroupVersion.APIResources {
-			resource := &resourceGroupVersion.APIResources[i]
-			// We don't want to check subObjects (like pods/status)
+			resource := &resourceGroupVersion.APIResources[i] // We don't want to check subObjects (like pods/status)
 			if len(strings.Split(resource.Name, "/")) != 1 {
 				continue
 			}
 
 			keyAPI := fmt.Sprintf("%s/%s", resourceGroupVersion.GroupVersion, resource.Kind)
 			if _, ok := kubeAPIs[keyAPI]; !ok {
-				gv, err := schema.ParseGroupVersion(resourceGroupVersion.GroupVersion)
-				if err != nil {
-					log.Fatalf("Failed to Parse GroupVersion of Resource: %s", err)
-				}
-				gvr := schema.GroupVersionResource{Group: gv.Group, Version: gv.Version, Resource: resource.Name}
-				list, err := dynClient.Resource(gvr).List(context.TODO(), metav1.ListOptions{})
-				if apierrors.IsNotFound(err) || apierrors.IsMethodNotSupported(err) {
-					continue
-				}
 
-				if apierrors.IsForbidden(err) {
-					log.Fatalf("Failed to list Server Resources of type %s/%s/%s. Permission denied! Please check if you have the proper authorization", gv.Group, gv.Version, resource.Kind)
-				}
+				gvr, list := getResources(dynClient, resourceGroupVersion.GroupVersion, resource.Name, resource.Kind)
 
-				if err != nil {
-					log.Fatalf("Failed to List objects of type %s/%s/%s. \nError: %v", gv.Group, gv.Version, resource.Kind, err)
+				if keyAPI == v1beta1IngressApi {
```

Comment:
ok, loved it! My only suggestion here is instead we need to "grow" the code here for each item in the cluster, do something more generic, like:
* create a map[string]string being the key the deprecated gvr string, and the value the new one
* check if keyAPI exists in the map, if so call "fixDeletedObjectList" passing the client, the key and the value
* Turn fixDeletedObjectList into something more generic

This way, in case of new false positives we can just grow this map instead of create new functions

wdyt?

Reply:
Hello Rikatz,

Your addition is really great, but if you don't mind, in my humble opinion, we can delay this refactoring until another concrete case appear. 

There's a chance that another case like that will never appear again and if appear, a refactoring doing exactly what you asked would be quick, clean and simple.

On top of that, by merging this PR as-is, we gonna be fixing a real issue with no further delays ;)

But if you prefer, I'll also be glad to implement your suggestions right now.

Please let me know wdyt ;)